### PR TITLE
Contract verify impl

### LINKED ISSUE #892: Add `verify` command to the cast tool
### Which component is your feature related to?

Cast

### Which component is your feature related to?

Cast

### Feature Request

Proposed design : https://github.com/foundry-rs/starknet-foundry/blob/master/design_documents/contract_verification.md

### PR DESCRIPTION
**Pull request by rjnrohit**
Closes #892 

## Introduced changes
This Pull Request adds the `verify` command for the contract verification. More about the details of the contract verification can found [here](https://github.com/foundry-rs/starknet-foundry/blob/master/design_documents/contract_verification.md).

- Added `verify command` to the `sncast` tool
- Modified `cast/main.rs` & created `cast/src/starknet_commands/verify.rs` mainly

## Checklist

- [x] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [x] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by drknzz — COMMENTED** [review]
Looks good 👍 could we have some tests for it?
Also, let's add later a mention in the docs and CHANGELOG.md about it

File: crates/cast/src/starknet_commands/verify.rs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    let verification_status = VerificationStatus::OK;
+    let errors = None;
+
+    let res = match verification_status {
+        VerificationStatus::OK => Ok(VerifyResponse {
+            verification_status,
+            errors,
+        }),
+        VerificationStatus::Error => Err(anyhow!("Unknown RPC error")),
+    };
```
[the remark this answers is not in the document]
Comment (rjnrohit) [reply crates/cast/src/starknet_commands/verify.rs:55]: This is just a dummy response for now until the voyager APIs become healthy.

File: crates/cast/src/starknet_commands/verify.rs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+use cast::helpers::response_structs::VerificationStatus;
+use cast::helpers::response_structs::VerifyResponse;
+use cast::helpers::scarb_utils::get_scarb_manifest;
+use clap::Args;
+use serde::Serialize;
+use starknet::core::types::FieldElement;
+use walkdir::WalkDir;
+
+#[derive(Args)]
+#[command(about = "Verify a contract through a block exporer")]
```
[the remark this answers is not in the document]
Comment (rjnrohit) [reply crates/cast/src/starknet_commands/verify.rs]: fixed, thanks!

File: crates/cast/src/starknet_commands/verify.rs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    #[clap(short = 'a', long)]
+    pub contract_address: FieldElement,
+
+    /// Name of the contract that is being verified
+    #[clap(short, long)]
+    pub contract_name: String,
+
+    /// Block explorer to use for the verification
+    #[clap(short = 'v', long = "verifier")]
+    pub verifier: String,
```
[the remark this answers is not in the document]
Comment (rjnrohit) [reply crates/cast/src/starknet_commands/verify.rs]: Added, thanks 🙂

File: crates/cast/src/starknet_commands/verify.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (rjnrohit) [reply crates/cast/src/starknet_commands/verify.rs:55]: done, thank you :)

File: crates/cast/src/starknet_commands/verify.rs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+    /// The network on which block explorer will do the verification
+    #[clap(short = 'n', long = "network")]
+    pub network: String,
```
[the remark this answers is not in the document]
Comment (rjnrohit) [reply crates/cast/src/starknet_commands/verify.rs]: done, thanks you:)

**Comment by rjnrohit** [comment]
> Looks good 👍 could we have some tests for it? Also, let's add later a mention in the docs and CHANGELOG.md about it

Will add tests when voyager APIs will become healthy and functional

**Review by stranger80 — APPROVED** [review]
LGTM, we now wait until Voyager Verififcation APIs are available for integration test.

**Review by THenry14 — CHANGES_REQUESTED** [review]
The code's looking very good, thank for the contribution 🔥 

A couple of things though:

Do you think it is in any way test-able? It would be cool to have any form of tests (even with mocks), this would make maintaining this code a lot easier 🙏 

Can you also add an info about this addition to CHANGELOG.md, and update the docs as well, please? There should be [appendix](https://github.com/foundry-rs/starknet-foundry/tree/master/docs/src/appendix/cast) and some [general overview with examples](https://github.com/foundry-rs/starknet-foundry/tree/master/docs/src/starknet) added for `verify`. Thanks!

**Pull request closed without merging**